### PR TITLE
Allow taking screenshot on web with canvasKit

### PR DESCRIPTION
**Pull request by passsy**
- Allow taking a screenshot on web with canvaskit
- Also added an error state because we don't use the retry mechanism on web
- Improved bottomsheet resizing with AnimatedSize

Screen-Shot-2021-03-14-23-15-45 31

### CODE REVIEW

File: lib/src/feedback/components/error_component.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        children: [
+          const SizedBox(height: 8),
+          ListTileButton(
+            key: const ValueKey('wiredash.sdk.retry_button'),
+            icon: WiredashIcons.redo,
+            iconColor: const Color(0xff9c4db1),
+            iconBackgroundColor: const Color(0xffffc4f0),
+            title: WiredashLocalizations.of(context)!.feedbackSubmitRetryTitle,
+            subtitle: WiredashLocalizations.of(context)!.feedbackSubmitRetryMsg,
+            onPressed: () => onRetry(),
```
Comment (atoka93) [inline lib/src/feedback/components/error_component.dart]: You can pass `onRetry` directly.

File: lib/src/feedback/components/error_component.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 16),
+      child: Column(
+        children: [
+          const SizedBox(height: 8),
+          ListTileButton(
+            key: const ValueKey('wiredash.sdk.retry_button'),
+            icon: WiredashIcons.redo,
+            iconColor: const Color(0xff9c4db1),
+            iconBackgroundColor: const Color(0xffffc4f0),
```
Comment (atoka93) [inline lib/src/feedback/components/error_component.dart]: Should we use the theme data here?

File: lib/src/feedback/components/loading_component.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 16),
+      child: Column(
+        children: const [
+          SizedBox(height: 32),
+          CircularProgressIndicator(),
```
Comment (atoka93) [inline lib/src/feedback/components/loading_component.dart]: Could the whole LoadingComponent be the `CircularProgressIndicator` wrapped with a `Padding`?

File: lib/src/feedback/components/error_component.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  const ErrorComponent({Key? key, required this.onRetry}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 16),
+      child: Column(
+        children: [
+          const SizedBox(height: 8),
```
Comment (atoka93) [inline lib/src/feedback/components/error_component.dart]: If there's already a `Padding` this could be moved there and the `Column` could be removed.

**Review by atoka93 — COMMENTED** [review]
Looking good 👍 

A few questions:

File: lib/src/feedback/components/error_component.dart (hunk 2 above)
Comment (passsy) [reply lib/src/feedback/components/error_component.dart]: Good idea, I haven't thought about the dark theme. We're not using the theme for the other icons, the colors luckily work with all themes :)

**Comment by passsy** [comment]
Keeping the branch for a while in case someone depends on it

**Pull request merged**